linalg/hypre.cpp

```diff
@@ -706,7 +707,7 @@ void HypreParMatrix::WrapHypreParCSRMatrix(hypre_ParCSRMatrix *a, bool owner)
    HypreRead();
 }
 
-signed char HypreParMatrix::CopyCSR(SparseMatrix *csr,
+signed char HypreParMatrix::CopyCSR(const SparseMatrix *csr,
```

Comment:
In this case `const` is actually wrong -- when we use shallow copy in `CopyMemory`, the contents of the `*src` is exposed in `mem_csr`.

The code compiles without errors because the methods `GetMemoryI`, `GetMemoryJ`, and `GetMemoryData` actually "strip" the const-ness of their contents -- they return `const Memory<T> &` but (as mentioned above) that does not actually protect the contents.

Reply:
Just to clarify: I think we should revert this change.

Reply:
Yeah, for some reason, it seemed to me it is only deep copying 🤔 . I reverted this in https://github.com/mfem/mfem/pull/4659/commits/193f8a68012d12772c179d01ba85df2dfedcb2e0 .